### PR TITLE
add functionality to wait for docker exec processes

### PR DESCRIPTION
**Pull request by jsingle**
@thieman I feel like if we're setting timeouts, we should let the calling test say what the timeout should be

### CODE REVIEW

File: tests/integration/cli/scripts_test.py [hunk 1]
```diff
@@ -33,13 +33,16 @@ def tearDown(self):
     def test_basic(self):
         self.assertFileNotInContainer('appa', '/app/a/foo')
         self.run_command('scripts appa example')
+        self.wait_for_exec_docker(1)
```
Comment (thieman) [inline tests/integration/cli/scripts_test.py]: I think it makes sense to provide a reasonable default, something like 3 seconds. If a test is going longer than that it's probably a crappy test, hopefully seeing someone override the default timeout would tip off a reviewer that there's a problem.

File: tests/testcases.py [hunk 2]
```diff
@@ -177,6 +176,17 @@ def run_command(self, args, fake_exit, raise_on_error=True):
             self._clear_stdout()
             return result
 
+    def wait_for_exec_docker(self, timeout):
+        def wait_for_processes():
+            for p in self.exec_docker_processes:
+                if p.returncode is None:
+                    p.communicate()
```
Comment (thieman) [inline tests/testcases.py]: Does the `returncode` get set even if you haven't called any follow-up functions like `communicate` on your `Popen` object?

File: tests/integration/cli/scripts_test.py (hunk 1 above)
Comment (jsingle) [inline tests/integration/cli/scripts_test.py]: sounds good

**Comment by thieman** [comment]
LGTM

File: tests/testcases.py (hunk 2 above)
Comment (jsingle) [inline tests/testcases.py]: Hmm I should really be using `poll` here

**Comment by jsingle** [comment]
Well this is working locally, but it sure isn't in jankins :rage:

**Pull request merged**